### ISSUE TITLE
Game crash when adding certain mods to skinned weapons

### ISSUE THREAD

**Issue by VictorKruger** [8fn/LegendarySkins-PD2#10]
Using the Legendary Skins mod not only unlocks the skins but they also unlock mods such as Barrel Extensions.  This allows the ability to add mods to skinned weapons that are not supposed to have a different mod attached.  Attempting to add certain mods to skinned weapons can lead to the game crashing.

My exact issue is with the secondary weapon, "The Judge Shotgun".  After applying the legendary skin "Anarcho" I attempted to add "The Silent Killer Suppressor" which locked up the game for a moment and then crashed.  

The workaround is using the mod once and then when all the skins are unlocked, remove the mod which properly sets back the weapon mod logic on skinned weapons to normal.

Script Stack Error:
Current thread: Main
Script stack:
deep_clone()  core/lib/utils/coretable.lua:34                     
_part_data()  lib/managers/weaponfactorymanager.lua:736           
_preload_part()  lib/managers/weaponfactorymanager.lua:372           
preload_blueprint()  lib/managers/weaponfactorymanager.lua:314           
preload_weapon_blueprint()  lib/managers/blackmarketmanager.lua:2115            
view_weapon()  lib/managers/blackmarketmanager.lua:5233            
_on_modified_weapon()  lib/managers/blackmarketmanager.lua:5197            
modify_weapon()  lib/managers/blackmarketmanager.lua:5150            
buy_and_modify_weapon()  lib/managers/blackmarketmanager.lua:5160            
clbk()  lib/managers/menu/blackmarketgui.lua:15178          
callback_func()  lib/managers/menu/blackmarketgui.lua:15545          
button_pressed()  lib/managers/dialogs/dialog.lua:61                  
button_pressed()  lib/managers/dialogs/genericdialog.lua:375          
button_pressed_callback()  lib/managers/dialogs/genericdialog.lua:371          
mouse_press()  lib/managers/dialogs/genericdialog.lua:123          
lib/managers/mousepointermanager.lua:345            

-------------------------------

**Comment by tskxz** [comment]
Try this
https://modworkshop.net/mod/25474

**Comment by loonesy** [comment]
My English is bad, but I want to ask with this mod to which you left a link, will I be able to modify legendary weapons?

**Comment by tskxz** [comment]
> My English is bad, but I want to ask with this mod to which you left a link, will I be able to modify legendary weapons?

Yes, u are able to apply modifications for some weapons, but some modifications cause crashs.

**Comment by loonesy** [comment]
> > My English is bad, but I want to ask with this mod to which you left a link, will I be able to modify legendary weapons?
> 
> Yes, u are able to apply modifications for some weapons, but some modifications cause crashs.

Ok , Thx
